code/modules/projectiles/guns/magic/staff.dm

```diff
@@ -1,6 +1,8 @@
 /obj/item/gun/magic/staff
 	slot_flags = ITEM_SLOT_BACK
 	ammo_type = /obj/item/ammo_casing/magic/nothing
+	worn_icon_state = null
+	worn_icon = 'icons/mob/clothing/back.dmi'
```

Comment:
this is already handled by `build_worn_icon` via the slot flag, remove it.

Reply:
Reverted. Had the worn_icon set to make the other slots use a working icon. I'll add it to the list of missing textures.